lib/ansible/module_utils/known_hosts.py

```diff
@@ -71,10 +72,15 @@ def is_ssh_url(url):
     return False
 
 def get_fqdn(repo_url):
+    return get_fqdn_and_port(repo_url)[0]
+
+def get_fqdn_and_port(repo_url):
 
     """ chop the hostname out of a url """
 
     result = None
+    port = SSH_KEYSCAN_DEFAULT_PORT
```

Comment:
We should not assume SSH port inside of this function.  (that way we'll be able to a more generic location later).  But we should be able to return a sentinel (like None) and then the caller can set the port to SSH_KEYSCAN_DEFAULT_PORT when that's the case.